### DISCUSSION TITLE
How to specify requests and limits for main container of orchestrator pod

### DISCUSSION THREAD

**Discussion by ogusak** [airbytehq/airbyte#37522] [Connector Questions]
- We are running Airbyte open source on EKS cluster with container orchestrator enabled (`CONTAINER_ORCHESTRATOR_ENABLED=true`). 
- We know we can control CPU and memory limits of main container of all job types (orchestor, source/destination connector) with worker's environment variable `JOB_MAIN_CONTAINER_CPU_LIMIT` (for CPU limit). 
- We'd like to control cpu and memory limits for the orchestrator container separately from other job pods as we noticed its main container requires a lot more CPU than any other job container (evident by monitoring CPU throttling metric for the container).
- We could not find any documentation on this issue and in the code could only find [this parameter](https://github.com/airbytehq/airbyte-platform/blob/bab15fa6c4343a4ea6718b8986dcb51453d2581d/airbyte-workers/src/main/resources/application.yml#L162) `REPLICATION_ORCHESTRATOR_CPU_LIMIT` 
- We provided `REPLICATION_ORCHESTRATOR_CPU_LIMIT` environment variable to worker pod/container but it didn't change the value of the CPU limit of the newly spawned orchestrator containers.

Please let us know if it is possible to control orchestrator's main container CPU limit.
Thanks!

**Comment by Vadym5** [comment]
the same questions..

**Comment by LarsHanegraaf** [comment]
I'm also interested in the answer, as it seems it is the performance bottleneck for our connectors built with the Connector Builder as well.

**Reply by Vadym5** [reply]
we resolved this via k8 admission controller with providing custom resources for pods we need

**Reply by LarsHanegraaf** [reply]
In the end I managed to get the result by altering the `values.yaml` of our Airbyte Kubernetes deployment:
```yaml
global:
  workload-launcher:
    resources:
      limits:
        cpu: 2
```

[a comment by an automated account is not shown]
